src/core/strategies/token.js

```diff
@@ -0,0 +1,285 @@
+ */
+function replaceByNegativeSign(token) {
+    const { offset } = token;
+    const { value } = terms.SUB;
+    return {
+        offset,
+        length: token.value.length,
+        value,
+        move: value.length - token.value.length
+    };
+}
+
+/**
+ * Produces a descriptor to replace a sign by a positive sign
+ * @param {token} token
+ * @param {number} index
+ * @param {token[]} tokens
+ * @returns {tokenChange}
+ */
+function replaceByPositiveSign(token, index, tokens) {
+    const allowExplicit = index && acceptExplicitPositive(tokens[index - 1]);
+    const { offset } = token;
+    const value = (allowExplicit && terms.ADD.value) || '';
+    return {
+        offset,
+        length: token.value.length,
+        value,
+        move: value.length - token.value.length
+    };
+}
+
+/**
+ * Apply a sign change at the current index
+ * @param {number} index - The index of the current token
+ * @param {token[]} tokens - The list of tokens that represent the expression
+ * @returns {tokenChange|null} - The result of the strategy: `null` if cannot apply, or the descriptor of the change
+ */
+function applySignChange(index, tokens) {
+    const token = tokens[index] || null;
+    const nextToken = tokens[index + 1] || null;
+    const type = tokensHelper.getType(token);
+    let result = null;
+
+    if (token) {
+        if (tokensHelper.isOperator(type)) {
+            // an operator precedes the operand
+            if (token.type === 'SUB' || token.type === 'NEG') {
+                // the operator is -, simply replace it by +
+                result = replaceByPositiveSign(token, index, tokens);
+            } else if (token.type === 'ADD' || token.type === 'POS') {
+                // the operator is +, simply replace it by -
+                result = replaceByNegativeSign(token);
+            } else if (nextToken) {
+                // the operator is not + or -, simply negate the value
+                result = insertNegativeSign(nextToken);
+            }
+        } else if (nextToken && (tokensHelper.isFunction(type) || token.type === 'LPAR')) {
+            // a function or a left parenthesis precedes the operand, simply negate the operand
+            result = insertNegativeSign(nextToken);
+        }
+    }
+
+    return result;
+}
+
+/**
+ * List of known strategies to apply on an expression in order to process sign change.
+ * Each strategy will return either `null` if cannot apply, or the descriptor of the change to apply.
+ * @type {tokenStrategy[]}
+ */
+export const signStrategies = [
+    /**
+     * Strategy that applies on numeric operands only
+     * @param {number} index - The index of the current token
+     * @param {token[]} tokens - The list of tokens that represent the expression
+     * @returns {tokenChange|null} - The result of the strategy: `null` if cannot apply, or the descriptor of the change
+     */
+    function strategyNumeric(index, tokens) {
+        let token = tokens[index] || null;
```

Comment:
im seeing this three lines quite a few times already, maybe refactoring could be necessary? :)

Reply:
well, not sure the refactoring would be better as this is only variable declarations...